aws/sdk/examples/dynamodb/src/bin/crud.rs

```diff
@@ -0,0 +1,452 @@
+use smithy_types::retry::RetryKind;
+
+use structopt::StructOpt;
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::fmt::SubscriberBuilder;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// Whether to run in interactive mode (you have to press return between operations)
+    #[structopt(short, long)]
+    interactive: bool,
+
+    /// The region
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// Activate verbose mode    
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Create a random, n-length string
+fn random_string(n: usize) -> String {
+    let mut rng = thread_rng();
+    iter::repeat(())
+        .map(|()| rng.sample(Alphanumeric))
+        .map(char::from)
+        .take(n)
+        .collect()
+}
+
+/// Create a new table. It's remotely possible the random table name exists.
+async fn create_table(client: &dynamodb::Client, table: &str, key: &str) {
+    let ad = AttributeDefinition::builder()
+        .attribute_name(key)
+        .attribute_type(ScalarAttributeType::S)
+        .build();
+
+    let ks = KeySchemaElement::builder()
+        .attribute_name(key)
+        .key_type(KeyType::Hash)
+        .build();
+
+    let pt = ProvisionedThroughput::builder()
+        .read_capacity_units(10)
+        .write_capacity_units(5)
+        .build();
+
+    match client
+        .create_table()
+        .table_name(table)
+        .key_schema(ks)
+        .attribute_definitions(ad)
+        .provisioned_throughput(pt)
+        .send()
+        .await
+    {
+        Ok(_) => println!(),
+        Err(e) => {
+            println!("Got an error creating the table:");
+            println!("{}", e);
+            process::exit(1);
+        }
+    }
+}
+
+/// For add_item and scan_item
+#[derive(Clone)]
+struct Item {
+    table: String,
+    key: String,
+    value: String,
+    first_name: String,
+    last_name: String,
+    age: String,
+    utype: String,
+}
+
+/// Add an item to the table.
+async fn add_item(client: &dynamodb::Client, item: Item) {
+    let user_av = AttributeValue::S(item.value);
+    let type_av = AttributeValue::S(item.utype);
+    let age_av = AttributeValue::S(item.age);
+    let first_av = AttributeValue::S(item.first_name);
+    let last_av = AttributeValue::S(item.last_name);
+
+    match client
+        .put_item()
+        .table_name(item.table)
+        .item(item.key, user_av)
+        .item("account_type", type_av)
+        .item("age", age_av)
+        .item("first_name", first_av)
+        .item("last_name", last_av)
+        .send()
+        .await
+    {
+        Ok(_) => println!(),
+        Err(e) => {
+            println!("Got an error adding item to table:");
+            println!("{}", e);
+            process::exit(1);
+        }
+    }
+}
+
+/// Scan the table for an item matching the input values.
+async fn scan(client: &dynamodb::Client, item: Item) {
+    let user_av = AttributeValue::S(item.value);
```

Comment:
is this a pattern we want to encourage users to use? shouldn't they use a dynamodb db query for this?

Reply:
I've replaced scan with query. Not sure I got the borrowing part right in the function.